### DISCUSSION TITLE
How to create a directive/middleware that applies to all routes

### DISCUSSION THREAD

**Discussion by develerltd** [99designs/gqlgen#2295] [Q&A]
### What happened?
I want to wrap a "needs auth by default" directive around the whole gqlgen resolver *but* need to be able to override it with directives like @public to allow for some resolvers to not need auth. Unable to do so.

### What did you expect?
To be able to apply a directive on QUERY and MUTATION that can't be removed or used by the gql client (like we have FIELD and FIELD_DEFINITION). It seems that any directive added to QUERY or MUTATION can only be added by the client.

**Comment by mpieczaba** [comment]
Well, I believe it can be achieved with [modelgen hook](https://gqlgen.com/recipes/modelgen-hook/), although this solution seems to be a little bit nasty. Global directives is a feature I look forward to see in next releases.
